http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -1767,17 +1772,27 @@ private void traceChunk(ByteBuf content) {
 
     private void traceHeaders(HttpHeaders headers) {
         for (String name : headers.names()) {
+            boolean isMasked = HEADER_MASK_PATTERNS.get().matcher(name).matches();
             List<String> all = headers.getAll(name);
             if (all.size() > 1) {
                 for (String value : all) {
-                    log.trace("{}: {}", name, value);
+                    String maskedValue = isMasked ? mask(value) : value;
+                    log.trace("{}: {}", name, maskedValue);
                 }
             } else if (!all.isEmpty()) {
-                log.trace("{}: {}", name, all.get(0));
+                String maskedValue = isMasked ? mask(all.get(0)) : all.get(0);
+                log.trace("{}: {}", name, maskedValue);
             }
         }
     }
 
+    private String mask(String value) {
+        if (value == null) {
+            return null;
+        }
+        return "*MASKED* [" + value.length() + " chars]";
```

Comment:
Subtle thing: I would not return the length. The fewer clues for attackers, the better :)

Reply:
I couldn't decide...  Then my pet project, I couldn't tell if the value was being injected into an auth header...

Showing the length let me see it was `Bearer ....some long token...` and not `Bearer {github.token}`

I agree though, as this feels convenient, and convenient things are usually security holes

I'll push a commit to remove it

Reply:
Done https://github.com/micronaut-projects/micronaut-core/pull/8336/commits/92b9c7208b11c1cef4388d66ff9fe7d1d72a5364